src/Castle.Core/DynamicProxy/IInterceptorBase.cs

```diff
@@ -0,0 +1,56 @@
+﻿
+namespace Castle.DynamicProxy
+{
+	using System;
+
+	public class ExceptionContext
+	{
+
+		public Exception Ex { get; set; }
+		public IInvocation Invocation { get; set; }
+	}
+
+	public abstract class InterceptorBase : IInterceptor
+	{
+		protected virtual void OnExcuting(IInvocation invocation)
+		{
+		}
+
+		protected virtual void OnExcuted(IInvocation invocation)
+		{
+		}
+
+		protected virtual void OnException(ExceptionContext context)
+		{
+		}
+
+		private bool NotVoidHasValue(IInvocation invocation)
+		{
+			return !(invocation.Method.ReturnType == typeof(void) ||
+			         invocation.ReturnValue == null);
```

Comment:
What if a non-`void` method *wants to* return `null`? It's a perfectly valid value, after all.

Reply:
Hi @stakx 
In my opinion
`invocation.ReturnValue` initial is `null`, developer set `invocation.ReturnValue`  that mean we don't want to execute this method (like do authorize).

thank you.